### PR TITLE
Add query package

### PR DESCRIPTION
**Pull request by stephen-soltesz**
This change add a new package to support running a BigQuery query with a parameterized row type and collect all results.

This change also adds basic CI configuration in travis and coveralls.

- - -
This change is [Reviewable](https://reviewable.io/reviews/m-lab/archive-repacker/1)

### CODE REVIEW

**Review by cristinaleonr — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/m-lab/archive-repacker/1)* status: 0 of 1 approvals obtained (waiting on @stephen-soltesz)

___
*[`query/query.go` line 18 at r3](https://reviewable.io/reviews/m-lab/archive-repacker/1#-NQl4uBu4SIeUWGCZYFw:-NQl4uBu4SIeUWGCZYFx:b-ouv43d) ([raw file](https://github.com/m-lab/archive-repacker/blob/01097f4547685e6435a4fbdddf4dd4480361e024/query/query.go#L18)):*
> ```Go
> var (
> 	// RowDelay pauses query result collection on every row.
> 	RowDelay = time.Duration(0)
> ```

Does this do anything if it's set to 0?
___
*[`query/query.go` line 46 at r3](https://reviewable.io/reviews/m-lab/archive-repacker/1#-NQl5Q0u1bz7LV1ArgHc:-NQl5Q0u1bz7LV1ArgHd:bbv3c2g) ([raw file](https://github.com/m-lab/archive-repacker/blob/01097f4547685e6435a4fbdddf4dd4480361e024/query/query.go#L46)):*
> ```Go
> 	t := time.Now()
> 
> 	var queryParams []bigquery.QueryParameter
> ```

Alternatively, instead of doing this conversion, the `params` parameter type could be `[]bigquery.QueryParameter`.
___
*[`query/query.go` line 63 at r3](https://reviewable.io/reviews/m-lab/archive-repacker/1#-NQl5tVp9mTGWhscs8Tw:-NQl5tVp9mTGWhscs8Tx:b-fx0gmx) ([raw file](https://github.com/m-lab/archive-repacker/blob/01097f4547685e6435a4fbdddf4dd4480361e024/query/query.go#L63)):*
> ```Go
> 		return nil, err
> 	}
> 	results := make([]Row, 0, 1000)
> ```

Where does the 1000 number come from?
___
*[`query/query.go` line 72 at r3](https://reviewable.io/reviews/m-lab/archive-repacker/1#-NQl5zNx6oWijfYi_ezD:-NQl5zNx6oWijfYi_ezE:b-785iph) ([raw file](https://github.com/m-lab/archive-repacker/blob/01097f4547685e6435a4fbdddf4dd4480361e024/query/query.go#L72)):*
> ```Go
> 		}
> 		results = append(results, row)
> 		time.Sleep(RowDelay)
> ```

Why do we need a delay?

**Review by stephen-soltesz — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/m-lab/archive-repacker/1)* status: 0 of 1 approvals obtained (waiting on @cristinaleonr)

___
*[`query/query.go` line 18 at r3](https://reviewable.io/reviews/m-lab/archive-repacker/1#-NQl4uBu4SIeUWGCZYFw:-NQlC0iHB80KII8szZps:bw2e5b9) ([raw file](https://github.com/m-lab/archive-repacker/blob/01097f4547685e6435a4fbdddf4dd4480361e024/query/query.go#L18)):*
<details><summary><i>Previously, cristinaleonr (Cristina Leon) wrote…</i></summary><blockquote>

Does this do anything if it's set to 0?
</blockquote></details>

Not when it's set to zero. The caller may want to set this to non-zero. 
___
*[`query/query.go` line 46 at r3](https://reviewable.io/reviews/m-lab/archive-repacker/1#-NQl5Q0u1bz7LV1ArgHc:-NQlDaA-3qUB_PO3Wj3Q:b-f5mcnb) ([raw file](https://github.com/m-lab/archive-repacker/blob/01097f4547685e6435a4fbdddf4dd4480361e024/query/query.go#L46)):*
<details><summary><i>Previously, cristinaleonr (Cristina Leon) wrote…</i></summary><blockquote>

Alternatively, instead of doing this conversion, the `params` parameter type could be `[]bigquery.QueryParameter`.
</blockquote></details>

The caller wouldn't need to know about any `bigquery` package types. But this is a little simpler still. Done.
___
*[`query/query.go` line 63 at r3](https://reviewable.io/reviews/m-lab/archive-repacker/1#-NQl5tVp9mTGWhscs8Tw:-NQlDbjiDY62cFg9qhw2:b-mxiver) ([raw file](https://github.com/m-lab/archive-repacker/blob/01097f4547685e6435a4fbdddf4dd4480361e024/query/query.go#L63)):*
<details><summary><i>Previously, cristinaleonr (Cristina Leon) wrote…</i></summary><blockquote>

Where does the 1000 number come from?
</blockquote></details>

Just a head start on the memory needed to read many results. It's not critical. Removed.
___
*[`query/query.go` line 72 at r3](https://reviewable.io/reviews/m-lab/archive-repacker/1#-NQl5zNx6oWijfYi_ezD:-NQlDyEa-4sDTVrM_RWb:b-o0m447) ([raw file](https://github.com/m-lab/archive-repacker/blob/01097f4547685e6435a4fbdddf4dd4480361e024/query/query.go#L72)):*
<details><summary><i>Previously, cristinaleonr (Cristina Leon) wrote…</i></summary><blockquote>

Why do we need a delay?
</blockquote></details>

The caller that will set the delay will have an explanation.

Basically, there is a maximum bytes/sec query read rate quota on BigQuery. This delay is a dial to slow things down across many clients if needed.

```
	// Many clients running a query can generate quota exceeded errors.
	// e.g. googleapi: Error 403: Quota exceeded: Your project exceeded
	//      quota for tabledata.list bytes per second per project.
	//
	// The aggregate quota is ~500Mbps.
	// * For 64 clients, that's < 8Mbps each.
	// * If each row is ~1k a 1ms pause is about 1Mbps
	// * 500us will add ~0.5s per 1000 results.
	// * ~128Mbps aggregate for 64 clients.

```

**Review by cristinaleonr — APPROVED** [review]
:lgtm: 

> *[Reviewable](https://reviewable.io/reviews/m-lab/archive-repacker/1#-:-NQlG49b4UoZ-vpqyFKB:bg7igkv)* status: :shipit: complete! 1 of 1 approvals obtained

**Comment by stephen-soltesz** [comment]
Thank you!

**Comment by stephen-soltesz** [comment]
Part of:
* https://github.com/m-lab/uuid-annotator/issues/48
